scripts/lib/mkcloud-dirmaint.sh

```diff
@@ -17,3 +17,46 @@ function dirmaint_do_cleanup()
 {
     echo "FIXME: do_cleanup"
 }
+
```

Comment:
This commit is another awesome set of refactorings :+1: but combining them all into a single commit makes it too hard to review, so I'll just take it on trust that this is simply extracting out hypervisor-specific into the relevant files without modifying behaviour.